addons/fennek/config.cpp

```diff
@@ -0,0 +1,303 @@
+#include "script_component.hpp"
+
+#define GRAD_AUTHOR "Grendel"
+#define GRAD_SIDE 1
+#define GRAD_FACTION "BLU_F"
+#define GRAD_CREW "B_Soldier_F"
+
+class CfgPatches {
+    class ADDON {
+        name = QUOTE(ADDON);
+        author = "$STR_GRAD_EXTRAS_AUTHOR";
+        authors[] = {GRAD_AUTHOR};
+        url = "$STR_GRAD_EXTRAS_URL";
+        requiredVersion = 1.32;
+        requiredAddons[] = {QMAINPATCH, "a3_soft_f_beta_mrap_03"};
+        VERSION_CONFIG;
+        units[] = {
+            "grad_extras_fennek_wd",
+            "grad_extras_fennek_wd_hmg",
+            "grad_extras_fennek_wd_gmg",
+            "grad_extras_fennek_wd_san",
+            "grad_extras_fennek_des",
+            "grad_extras_fennek_des_hmg",
+            "grad_extras_fennek_des_gmg",
+            "grad_extras_fennek_des_san",
+            "grad_extras_fennek_snow",
+            "grad_extras_fennek_snow_hmg",
+            "grad_extras_fennek_snow_gmg",
+            "grad_extras_fennek_snow_san",
+            "grad_extras_fennek_un",
+            "grad_extras_fennek_un_san"
+        };
+        weapons[] = {};
+    };
+};
+
+class grad_extras_fennek_TextureSources {
```

Comment:
Was hat es auf sich mit der Classe außerhalb der Standard Classen (CfgVhicles, CfgWeaons) ?

Reply:
So können alle Fenneks die gleichen `TextureSources` haben ohne, dass sie vom gleichen Fahrzeug erben müssen. Sonst hätte ich ja alle texture sources jeweils drei mal (bei Unarmed, HMG und GMG) definieren müssen. Jetzt muss ich nur in Zeile [162](https://github.com/gruppe-adler/grad_extras/blob/33917fbe34a16435c38f317a9832658bee979f2e/addons/fennek/config.cpp#L162), [181](https://github.com/gruppe-adler/grad_extras/blob/33917fbe34a16435c38f317a9832658bee979f2e/addons/fennek/config.cpp#L181) und [200](https://github.com/gruppe-adler/grad_extras/blob/33917fbe34a16435c38f317a9832658bee979f2e/addons/fennek/config.cpp#L200) davon erben.

Is vielleicht ned die schönste Lösung aber, besser als alles dreifach zu schreiben.

Reply:
Ah, okay. Ne ist ne super Lösung. War mir nur nicht klar wofür das war.